fvm/transactionEnv.go

```diff
@@ -735,6 +738,24 @@ func (e *TransactionEnv) GenerateUUID() (uint64, error) {
 }
 
 func (e *TransactionEnv) meterComputation(kind common.ComputationKind, intensity uint) error {
+	// this method is called on every unit of operation, so
+	// checking the context here is the most likely would capture
+	// timeouts or cancellation as soon as they happen, though
+	// we might revisit this when optimizing script execution
+	// by only checking on specific kind of meterComputation calls.
+	//
+	// TODO: in the future this context check should be done inside the cadence
+	select {
+	case <-e.executionCtx.Done():
+		err := e.executionCtx.Err()
+		if errors.Is(err, context.DeadlineExceeded) {
+			return errors.NewExecutionTimedOutError()
+		}
```

Comment:
TxEnv should **not** support deadline through context, as it could result in non-deterministic execution on different ENs and result in execution forks. And since EN would almost never cancel the execution of a transaction, it would impact the performance a bit as well.  Any execution limitation outside of computation, storage interaction and memory metering currently is almost impossible in the current architecture.